src/main/java/org/team1540/robot2025/subsystems/drive/ModuleIOTalonFX.java

```diff
@@ -0,0 +1,255 @@
+        inputs.turnSupplyCurrentAmps = turnSupplyCurrent.getValueAsDouble();
+        inputs.turnStatorCurrentAmps = turnStatorCurrent.getValueAsDouble();
+        inputs.turnTempCelsius = turnTemp.getValueAsDouble();
+
+        // Update odometry inputs
+        inputs.odometryTimestamps =
+                timestampQueue.stream().mapToDouble(Double::doubleValue).toArray();
+        inputs.odometryDrivePositionsRads = drivePositionQueue.stream()
+                .mapToDouble(Units::rotationsToRadians)
+                .toArray();
+        inputs.odometryTurnPositions =
+                turnPositionQueue.stream().map(Rotation2d::fromRotations).toArray(Rotation2d[]::new);
+        timestampQueue.clear();
+        drivePositionQueue.clear();
+        turnPositionQueue.clear();
+    }
+
+    @Override
+    public void setDriveOpenLoop(double input) {
+        drive.setControl(
+                switch (constants.DriveMotorClosedLoopOutput) {
+                    case Voltage -> driveVoltageReq.withOutput(input);
+                    case TorqueCurrentFOC -> driveTorqueReq.withOutput(input);
+                });
+    }
+
+    @Override
+    public void setTurnOpenLoop(double input) {
+        turn.setControl(
+                switch (constants.SteerMotorClosedLoopOutput) {
+                    case Voltage -> turnVoltageReq.withOutput(input);
+                    case TorqueCurrentFOC -> turnTorqueReq.withOutput(input);
+                });
+    }
+
+    @Override
+    public void setDriveVelocity(double velocityRadPerSec) {
+        double velocityRotPerSec = Units.radiansToRotations(velocityRadPerSec);
+        drive.setControl(
+                switch (constants.DriveMotorClosedLoopOutput) {
+                    case Voltage -> driveVelocityVoltageReq.withVelocity(velocityRotPerSec);
+                    case TorqueCurrentFOC -> driveVelocityTorqueReq.withVelocity(velocityRotPerSec);
+                });
+    }
+
+    @Override
+    public void setTurnPosition(Rotation2d rotation) {
+        turn.setControl(
+                switch (constants.SteerMotorClosedLoopOutput) {
+                    case Voltage -> turnPositionVoltageReq.withPosition(rotation.getRotations());
+                    case TorqueCurrentFOC -> turnPositionTorqueReq.withPosition(rotation.getRotations());
+                });
+    }
+
+    @Override
+    public void setDriveBrakeMode(boolean enabled) {
+        if ((driveConfig.MotorOutput.NeutralMode == NeutralModeValue.Brake) == enabled
+                && lastDriveConfigStatus.isOK()) {
+            return;
+        }
+
+        brakeModeExecutor.execute(() -> {
+            synchronized (driveConfig) {
+                driveConfig.MotorOutput.NeutralMode = enabled ? NeutralModeValue.Brake : NeutralModeValue.Coast;
+                lastDriveConfigStatus =
+                        PhoenixUtil.tryUntilOk(5, () -> drive.getConfigurator().apply(driveConfig.MotorOutput, 0.25));
```

Comment:
So you can change the neutral mode of a spinning motor and it doesn't affect it? Please confirm this with docs or testing

Reply:
This has been tested, but it would definitely be beneficial to test this more. I did take this code from 6328 who used it all of last year without problems, however.

Reply:
Also neutral mode only an impact on motor behavior when it's outputting 0V, so changing neutral mode while spinning shouldn't affect anything.

Reply:
In theory sure but it depends on how their firmware deals with config applications. It could be blocking motor-side, in which case PID loops won't recalculate until it's done.